public/resources/core/js/services/billing-account.service.js

```diff
@@ -0,0 +1,162 @@
+            return this.customerAccount.customer.getTelephoneNumber();
+        };
+        BillingAccount.prototype.serialize = function serialize() {
+            return {
+                id: this.id,
+                href: this.href
+            };
+        };
+
+        var EVENTS = {
+        };
+
+        var TYPES = {
+        };
+
+        var TEMPLATES = {
+            BILLING_ACCOUNT: {
+                name: '',
+                state: 'Defined',
+                customerAccount: {},
+                relatedParty: []
+            }
+        };
+
+        return {
+            EVENTS: EVENTS,
+            TEMPLATES: TEMPLATES,
+            TYPES: TYPES,
+            search: search,
+            create: create,
+            detail: detail,
+            launch: launch
+        };
+
+        function search(filters) {
+            var deferred = $q.defer();
+            var params = {};
+
+            BillingAccount.query(params, function (billingAccounts) {
+                detailCustomerAccount(billingAccounts).then(function () {
+                    deferred.resolve(billingAccounts);
+                }, function (argument) {
+                    deferred.reject(response);
+                });
+            }, function (response) {
+                deferred.reject(response);
+            });
+
+            return deferred.promise;
+        }
+
+        function create(data) {
+            var deferred = $q.defer();
+
+            Customer.create(data.customerAccount.customer).then(function (customer) {
+                data.customerAccount.customer = customer;
+                CustomerAccount.create(data.customerAccount).then(function (customerAccount) {
+                    var user = User.serializeBasic();
+                    user.role = 'bill receiver';
+                    var resource = angular.extend({}, data, {
+                        name: customerAccount.name,
+                        customerAccount: customerAccount.serialize(),
+                        relatedParty: [user]
+                    });
+
+                    BillingAccount.save(resource, function (billingAccount) {
+                        billingAccount.customerAccount = customerAccount;
+                        deferred.resolve(billingAccount);
+                    }, function (response) {
+                        deferred.reject(response);
+                    });
+                }, function (response) {
+                    deferred.reject(response);
+                });
+            }, function (response) {
+                deferred.reject(response);
+            });
+
+            return deferred.promise;
+        }
+
+        function detail(id) {
+            var deferred = $q.defer();
+            var params = {
+                id: id
+            };
+
+            BillingAccount.get(params, function (billingAccount) {
+                CustomerAccount.detail(billingAccount.customerAccount.id).then(function (customerAccount) {
```

Comment:
Write a comment here explaining that the Customer Account service will embed the information of the Customer...